Add a "keep-versions" parameter.

michaelkaye opened the pull request:
If set, tidy up disk after extraction to manage space.

Partially fixes #1

michaelkaye committed:
Add a "keep-versions" parameter.

If set, tidy up disk after extraction to manage space.

michaelkaye committed:
Tidy up

michaelkaye committed:
Clean up doc

richvdh commented on buildkite-webhook-listener.py:
```diff
@@ -195,8 +199,23 @@ def deploy_tarball(artifact_url, target_dir):
+    Will never remove the target_dir that we just deployed.
+    Will only consider directories that match the pattern.
+    """
+    directories = glob.glob(args_extract_directory + "/" + pipeline_name + "-#*")
```
is there any guarantee that the tarballs are called `pipeline_name-#*` ?

michaelkaye replied:
I was hoping that would be valid, but maybe we just look for `args_extract_directory + "/*-#*"` and try not to have two different pipelines unpacking in the same directory

richvdh replied:
> is there any guarantee that the tarballs are called pipeline_name-#* ?

answer: yes, because that is the name we construct at https://github.com/matrix-org/buildkite-webhook-listener/blob/3806783e7fff69c0503446d6fc44fb81ad550845/buildkite-webhook-listener.py#L161.

so I think the original was fine. I think the new version would also be fine, but `pipeline_name` is now redundant in `tidy_extract_directory`, so should be removed if you're going down that path.

richvdh commented on buildkite-webhook-listener.py:
```diff
@@ -248,13 +267,25 @@ def deploy_tarball(artifact_url, target_dir):
         )
     )
 
+    parser.add_argument(
```
I think you need `type=int` to (a) make argparse validate it as an int, (b) avoid an error when you try to apply unary `-` to a string later

michaelkaye committed:
Address code review.

Parse arguments correctly, and delete unpacked tarballs of any name in the extract folder.

richvdh commented on buildkite-webhook-listener.py:
```diff
@@ -268,4 +299,12 @@ def deploy_tarball(artifact_url, target_dir):
     )
+    if arg_keep_versions is not None:
+        print(
+            "Keeping only previous %s versions." % ( arg_keep_versions ),
```
Prefer to pass a tuple to %. `( foo )` is the same as `foo`. You need a trailing comma inside the parens to make it a tuple.

```suggestion
            "Keeping only previous %i versions." % (arg_keep_versions, ),
```

richvdh commented on buildkite-webhook-listener.py:
```diff
@@ -195,8 +199,23 @@ def deploy_tarball(artifact_url, target_dir):
+    
 
-
+def tidy_extract_directory(target_dir, pipeline_name):
```
it'd be nice to document what the parens mean; see under https://google.github.io/styleguide/pyguide.html#doc-function-raises for the style we follow in backendland.

michaelkaye replied:
I realised that since an earlier change, we also don't use pipeline_name for anything.

richvdh reviewed (approved):
this is basically fine. I've commented on a couple of nits but for a script this size they're hardly critical.

michaelkaye committed:
Update buildkite-webhook-listener.py

Co-authored-by: Richard van der Hoff <1389908+richvdh@users.noreply.github.com>

michaelkaye committed:
Remove 'pipeline_name' parameter and document.
